### ISSUE TITLE
FMU recompilation create_zip missing dependencies

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#10719]
### Description

When generating FMUs with the CMakeLists.txt there is a custom target `create_zip` that generates the FMU zip file.
It depends on the install target, but CMake doesn't know about this. So it could happen, that the FMU is zipped before all executable are compiled.

### Steps to Reproduce

Generate a CMake FMU, unzip it into `FMU/` and run

```
$ rm -rf FMU/binaries
$ cd FMU/sources
$ cmake -S . -B build_cmake 
$ cmake --build build_cmake/ --target create_zip install --parallel
```

### Expected Behavior

When calling `create_zip` is called, the `install` target needs to be run first.

### Version and OS

 - OpenModelica Version: v1.22.0-dev-102-g8fa5810e98-cmake
 - OS: Ubuntu Focal

**Comment by mahge** [comment]
Yes, unfortunatelly, this does not seem possible yet. Apparently, this was [requested](https://gitlab.kitware.com/cmake/cmake/-/issues/8438) 14 years ago and has not happened yet 🙂. So ya, we should probably go with the manual option.

**Comment by bilderbuchi** [comment]
I'm not totally sure what's all included in creating a zip file, but might https://cmake.org/cmake/help/latest/module/CPack.html be an option to package everything?

**Comment by mahge** [comment]
> I'm not totally sure what's all included in creating a zip file, but might https://cmake.org/cmake/help/latest/module/CPack.html be an option to package everything?

That is actually a very good idea and the one that seems to fit the intention.  We just have to figure out the details of when and how to call it. Ideally, what we need to do is something like:
```sh
cd <fmu_build_dir>
cpack -G ZIP
```
and it should create a zip containing anything that is _installed by cmake (using `install()`)_ . Maybe it is something we can check and see how easy it might be to do.

**Comment by bilderbuchi** [comment]
I think you should get an additional target  `package`/`package_source`, so you don't even have to invoke cpack separately: https://cmake.org/cmake/help/latest/module/CPack.html#targets-package-and-package-source

**Comment by AnHeuermann** [comment]
I don't have the time to check how CPack works just now. For my use-case it's enough to ensure, that the packed FMU has all files (and nothing is missing depending on whatever order CMake decides to do stuff in parallel).
However if someone can check out CPack and provide a snipped or PR we can integrate it into OM.

**Comment by AnHeuermann** [comment]
Closing this for now. If someone needs the CPack way we should open a new ticket.
